README.rst

```diff
@@ -262,6 +263,24 @@ foo
 bar
 
 
+Computing and Retrieving Checksums
+----------------------------------
+
+Each data object may be associated with a checksum by calling chksum() on the object in question.  Various 
+behaviors can be elicited by passing in combinations of keywords (for a description of which, please consult the
+`header documentation <https://docs.irods.org/4.2.10/doxygen/dataObjChksum_8h.html>`_ .)
+
+As with most other iRODS API's, it is straightforward to specify keywords by adding them to an option dictionary:
+
+>>> data_object_1.chksum()  # - computes the checksum if already in the catalog, otherwise computes and stores it
+...                         #   (ie. default behavior with no keywords passed in.)
+>>> import irods.keywords as kw
+>>> opts = { kw.VERIFY_CHKSUM_KW:'' }
+>>> data_object_2.chksum( **opts )  # - Uses verification option. (Does not auto-vivify a checksum field).
```

Comment:
is this "does not auto-vivify" note true?  verification will calculate one *unless* NO_COMPUTE_KW is passed...

Reply:
It does seem to be true.  Here's the script I'm playing with:
```
$ more trychksum.py 
#!/usr/bin/env python
from __future__ import print_function
import sys
import irods.keywords as kw
from getopt import getopt
options,args=getopt(sys.argv[1:],"K",["no-compute"])
import irods.test.helpers
with irods.test.helpers.make_session()as ses:
  d = ses.data_objects.get(args[0])
  opts={}
  for sw,val in options:
    if sw == '-K': 
      opts[kw.VERIFY_CHKSUM_KW]=''
    elif sw == '--no-compute':
      opts[kw.NO_COMPUTE_KW]=''
  print('chksum = %r' % d.chksum(**opts))
  

```
When I do the following
```
$ irm -fr newfile ; itouch newfile
./trychksum -K -- `ipwd`/a
```
It doesn't return or store a checksum. Not sure why.

Reply:
The same is true of -K (=>VERIFY_CHKSUM_KW) whether used with this script or with  `ichksum`.  It doesn't store or compute a checksum, independent of `--no-compute` so this appears consistent.

Reply:
ah, yes, `--verify` will compute and compare, but not store.    and it will skip the compute part (only size check) if `--no-compute` is passed.   okay, still good...